### PR TITLE
Add controller unit tests

### PR DESCRIPTION
**Pull request by MB-Finski**
Adds unit tests for all controller methods

### CODE REVIEW

**Comment by julien-nc** [comment]
Could you sign your commits with a GPG key?
It requires to choose a key and enable signing in your `~/.gitconfig` file and then make the commits with the `-s` option. For existing commits, you can sign them by rebasing like described in the DCO action output: 
```
git rebase HEAD~1 --signoff
git push origin dev -f
```

File: tests/unit/Controller/TemplateControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+				$userTemplates,
+				$adminTemplates
+			);
+
+		$controller = new TemplateController(
+			$this->appName,
+			$this->request,
+			$this->templateMapper,
+			$this->userId
+		);
```
Comment (julien-nc) [inline tests/unit/Controller/TemplateControllerTest.php:73]: You could instanciate this only once in the setUp method.

**Review by julien-nc — APPROVED** [review]
This is good as a base.
I think it would make sense to make "integration" tests since the app mostly access the database. In other words, you could really access the DB, so you could either stop mocking the templateMapper in this test file or have another test file testing the templateMapper.
Not urgent, do it if you feel like it or create an issue so we don't forget.

**Comment by julien-nc** [comment]
Oh also, you could add Phpunit as a composer dev dependency 
```
composer require --dev phpunit/phpunit=^9.0.0
```
and then in the last line of the phpunit.yml GH action, you can run phpunit with
```
run: vendor/bin/phpunit -c tests/phpunit.xml
```

**Comment by MB-Finski** [comment]
Oh no, sorry! I'd forgotten about this at some point!

**Pull request merged**